client/my-sites/stats/stats-module/index.jsx

```diff
@@ -141,7 +143,15 @@ class StatsModule extends Component {
 		} = this.props;
 
 		// Only show loading indicators when nothing is in state tree, and request in-flight
-		const isLoading = ! this.state.loaded && ! ( data && data.length );
+		let isLoading = ! this.state.loaded && ! ( data && data.length );
```

Comment:
I don't think that extending this component this way is a good idea. All modules using this file are fetching the data the same way. The new code would create a fork in that logic. This would cascade into having to perform a check in all crucial parts of the component, making two components meshed into one that is complex.

Creating a new component in the `stats-module` directory and using `StatsListCard` with your new fetching mechanism would probably make more sense. You also suspect we'll have to make more changes to reflect the designs; therefore, a new component seems even more appealing.

Reply:
That makes a lot of sense. I'll work on those changes today.

Reply:
I've introduced a new `StatsModule2` component which is basically just a copy of the existing `StatsModule` component minus the Redux bits (and it's not a class-based component). Was this what you had in mind @grzegorz-cp?

On the one hand, it feels like a lot of code duplication. On the other hand, it could allow us to slowly move the various modules away from the query component set up and over to React Query, at which point we could remove the old module.

Let me know your thoughts on this approach!

Reply:
Thanks for doing this Kevin. I think this is a good direction, but you are right that there was a lot of duplicated code. I reduced the number of props and renamed it. When we finish extending the component (with all of the custom elements like dropdown and header elements), we'll be able to see if we can extract the duplicated pieces in one shared component.